src/Behat/Behat/Context/Cli/InteractiveContextIdentifier.php

```diff
@@ -57,6 +57,10 @@ public function __construct(TranslatorInterface $translator, InputInterface $inp
      */
     public function guessTargetContextClass(ContextEnvironment $environment)
     {
+        if ($this->input->hasParameterOption('--no-interaction')) {
```

Comment:
Good catch, though. I made the tradeoff more explicit: https://github.com/Behat/Behat/commit/c04e67f3a1cfad72b7e79e192e537fb42268af26